ttnn/cpp/ttnn/operations/data_movement/transpose/device/transpose_program_factory.hpp

```diff
@@ -566,6 +566,293 @@ operation::ProgramWithCallbacks transpose_hc_multi_core(const Tensor &a, Tensor
     return {.program=std::move(program), .override_runtime_arguments_callback=override_runtime_args_callback};
 }
 
+
+operation::ProgramWithCallbacks transpose_hc_multi_core_sharded(const Tensor &a, Tensor &output) {
```

Comment:
If you touch this file again in another PR, could you separate it out into a .hpp with the declarations and the .cpp with the implementations

Reply:
the implementation is in ttnn/cpp/ttnn/operations/data_movement/transpose/device/transpose_program_factory.hpp
you want it to separate into another .cpp file?

Reply:
yes exactly, but you don't have to do it in this PR since it's not urgent.

Reply:
that means the transpose_program_factory.hpp should be gone right? since all the implementations are in there